basics/funktionen.tex

```diff
@@ -1,97 +1,50 @@
 \lesson{Funktionen}
 
-Wir haben den Begriff schon ein paarmal genannt und auch, wenn ihr es
-vielleicht nicht gemerkt habt, sie waren schon die ganze Zeit vor eurer Nase:
-Funktionen.
+Aus der Mathematik kennt ihr bereits Funktionen, wie zum Beispiel $f(x) = 4x^3 - 8x^2 + 16x - 12$.
+Eine wichtige Idee dahinter ist es einfach $f(3)$ zu schreiben, wenn man eigentlich $4 \cdot 3^3 - 8 \cdot 3^2 + 16 \cdot 3 - 12$ meint.
+Da dies häufig vorteilhaft ist, wurde diese Funktionalität in die meisten Programmiersprachen übernommen.
+
+In \Cpp können Funktionen Daten, die sogenannten \emph{Parameter}, entgegen nehmen, diese werden wie ganz normale Variablen behandelt, und in Abhängigkeit dieser verschiedene Ergebnisse zurückgeben, der sogenannte \emph{Rückgabewert}.
+
+An dieser Stelle ist der Unterschied zwischen Rückgabe und Ausgabe wichtig: Eine Ausgabe (gekennzeichnet durch \texttt{std::cout}) gibt Informationen auf dem Bildschirm für die Nutzerin aus, eine Rückgabe (gekennzeichnet durch \texttt{return}) gibt hingegen ein bestimmtes Ergebnis an einen anderen Teil des Programmes zurück, damit dieser dort in einer Variable gespeichert oder direkt weiter verarbeitet werden kann.
```

Comment:
Welches an dieser Stelle der richtige Ausdruck ist, hängt denke ich von deiner Perspektive ab: Wenn du selbst den Code schreibst *bestimmst* du was ausgegeben wird, wenn du Code liest beziehungsweise lernen willst zu bestimmen, ist *gekennzeichnet durch* denke ich angebrachter, da man in einer passiven Rolle ist.